File: client/common/model/gather/gather-sku.ts
export class GatherSku {
  constructor(
    public id: number,
    public batchId: number,
    public name: string,
    public source: string,
    public sales: string,
    public price: string,
    public skuId: string,
    public favorite: boolean,
    public createdAt: string,
    public createdBy: number,
    public updatedAt: string,
    public updatedBy: number
  ) {
    this.id = id
    this.batchId = batchId
    this.name = name
    this.source = source
    this.sales = sales
    this.price = price
    this.skuId = skuId
    this.favorite = favorite
    this.createdAt = createdAt
    this.createdBy = createdBy
    this.updatedAt = updatedAt
    this.updatedBy = updatedBy
  }
}

export class GatherSkuCreateReq {
  constructor(
    public batchId: number,
    public name: string,
    public source: string,
    public sales: string,
    public price: string,
    public skuId: string,
    public favorite: boolean
  ) {
    this.batchId = batchId
    this.name = name
    this.source = source
    this.sales = sales
    this.price = price
    this.skuId = skuId
    this.favorite = favorite
  }
}

File: server/internal/gather/models/gather_sku.go
package models

import (
	"server/common/middleware/database"
)

type GatherSku struct {
	database.BaseEntity
	BatchID  uint64 `json:"batchId"`
	Name     string `json:"name"`
	Source   string `json:"source"`
	Sales    string `json:"sales"`
	Price    string `json:"price"`
	SkuID    string `json:"skuId"`
	Favorite bool   `json:"favorite"`
}

func (s *GatherSku) TableName() string {
	return "gather_sku"
}
